datalad/utils.py

```diff
@@ -445,10 +445,13 @@ def rmtree(path, chmod_files='auto', children_only=False, *args, **kwargs):
     # Give W permissions back only to directories, no need to bother with files
     if chmod_files == 'auto':
         chmod_files = on_windows
-    # TODO:  yoh thinks that if we could quickly check our Flyweight for
-    #        repos if any of them is under the path, and could call .precommit
-    #        on those to possibly stop batched processes etc, we did not have
-    #        to do it on case by case
+
+    # Make sure *Repo instances that used to point to `path` are
+    # actually destructed before trying to delete anything. This is particularly
+    # important regarding git-annex batch processes that might not have been
+    # closed yet otherwise.
+    gc.collect()
```

Comment:
if we were to proceed with this (after checking the impact on performance through timing tests run and checking effect on our few benchmarks) -- might be worth consider addressing to be removed `TODO` comment - such check might be not that heavy (although might be shown to be heavier than just calling `gc.collect`)

Reply:
That check would be cheap - it's just looking up a key in a dict. There's no `precommit` to call anymore, though. What needs to be done is killing that object (it would be invalid after an `rmtree` anyway).

Anyways - this PR is discontinued, since we found a better solution.